plugins/samples/immediate_response/tests.textpb

```diff
@@ -0,0 +1,65 @@
+test {
+  name: "ImmediateResponseFromRequestHeaders"
```

Comment:
Similar feedback to PR#183: this sample + test doesn't help Service Extensions users get stuff done.

We have samples that use ImmediateResponse in headers:
https://github.com/search?q=repo%3AGoogleCloudPlatform%2Fservice-extensions%20sendLocalResponse&type=code

We should perhaps add a body-based ImmediateResponse sample, if we determine that is a good thing (TBD, see other comment). But we should write a sample for a real use case, such as a WAF that inspects request body. ImmediateResponse in a body handler should be a separate PR.